pages/learn/deploy/deploy-with-balena-button.md

```diff
@@ -21,18 +21,18 @@ __Note:__ Currently git submodules are not supported and will not build properly
 
 You can add the **Deploy with {{ $names.company.lower }}** button to any project that can be deployed to {{ $names.cloud.lower }}. To add the button to a project repository, add the following to, for example, the project repository's README.md file:
 
-`[![](https://www.balena.io/deploy.png)](https://dashboard.balena-cloud.com/deploy?repoUrl=<your-repo-url>)`
+`[![balena deploy button](https://www.balena.io/deploy.png)](https://dashboard.balena-cloud.com/deploy?repoUrl=<your-repo-url>)`
 
 Note that you can further customize the button's behavior through the use of [query string parameters](#query-string-parameters).
 
 ### Query string parameters
 
 You can further customize the behavior of the **Deploy with {{ $names.company.lower }}** button by providing additional URL parameters. The following URL parameters are available and may be appended to the `https://dashboard.balena-cloud.com/deploy` link:
 
-* `repoUrl` - The URL of the project repository.
+* `repoUrl` - The URL of the project repository. If you are placing the deploy button in a GitHub repo then {{ $names.cloud.lower }} can auto-determine the `repoUrl` from the referer info in the HTTP headers. However on Firefox and with some ad-blockers this may fail. So we recommend that you populate this query string parameter.
 * `tarballUrl` - The URL of the project tarball. Automatically determined from `repoUrl` if not provided.
 * `configUrl` - The URL of the configuration file of the application. Automatically determined from `repoUrl` if not provided.
-* `defaultDeviceType` - The device type that will be pre-selected in the "Create application" modal. It defaults to Raspberry Pi 4 if not provided.
+* `defaultDeviceType` - The device type that will be pre-selected in the "Create application" modal. It defaults to Raspberry Pi 4 if not provided. You can find a list of [device types here](reference/hardware/devices/).
```

Comment:
It might be ok without it, not sure.
```suggestion
* `defaultDeviceType` - The device type that will be pre-selected in the "Create application" modal. It defaults to Raspberry Pi 4 if not provided. You can find a list of [device types here](/reference/hardware/devices/).
```